administrator/components/com_media/Model/Api.php

```diff
@@ -120,23 +121,30 @@ public function getFile($adapter, $path = '/')
 	 * @param   string  $adapter  The adapter
 	 * @param   string  $path     The folder
 	 * @param   string  $filter   The filter
+	 * @param   array   $options  The options
 	 *
 	 * @return  \stdClass[]
 	 *
 	 * @since   __DEPLOY_VERSION__
 	 * @throws  \Exception
 	 * @see     AdapterInterface::getFile()
 	 */
-	public function getFiles($adapter, $path = '/', $filter = '')
+	public function getFiles($adapter, $path = '/', $filter = '', $options = array())
 	{
-		// Add adapter info to all the files to be returned
+		// Add adapter prefix to all the files to be returned
 		$files = $this->getAdapter($adapter)->getFiles($path, $filter);
 
 		foreach ($files as $file)
 		{
 			$file->path = $adapter . ":" . $file->path;
 		}
 
+		// Add url info to a file when option is set
+		if (in_array("url", $options) && count($files) === 1 && $files[0]->type == "file")
```

Comment:
even we added url for files in a directory if it contain a directory it makes no sense, IMO we should force them to apply this only on a single file. Since 3dps will never touch this we can make em force to avoid unnecessary checks in api